### ISSUE TITLE
Branch custom actions for Gerrit

### ISSUE THREAD

**Issue by cnovel** [ForkIssues/TrackerWin#301]
This is a very specific workflow when using Gerrit Code Review. The commit can't be pushed directly to the origin branch (let's say origin/master) but to a staging branch (origin/refs/for/master) where the commit is validated through the interface and then merged in origin master.

Now in my git config for my repo I have the following:
```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[submodule]
	active = .
[remote "origin"]
	url = ssh://cnovel@xx.xxx.xx.xx:29418/myrepo
	fetch = +refs/heads/*:refs/remotes/origin/*
	push = refs/heads/*:refs/for/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
```

But Fork doesn't seem to be taken this modification into account. When pushing the only option available is origin/master

![image](https://user-images.githubusercontent.com/5690282/58813472-26d5c800-8624-11e9-9565-b5ffaf83e421.png)

Now I'm no git expert so I'm not sure the origin branch should be origin/refs/for/master. I can provide you some logs from SourceTree (I'm trying to switch) if that's helping you.

**Comment by DanPristupov** [comment]
I think you can create a custom action for that.

You can find some info there: https://github.com/ForkIssues/Tracker/issues/123

**Comment by cnovel** [comment]
Is it possible to add a variable $branch in Commit Custom Action? That way my action can be $git push origin HEAD:refs/for/$branch, instead of creating a custom action per branch.

**Comment by DanPristupov** [comment]
We probably need a new type of custom action - Reference custom action, which could be applied to local and (or) remote branches and (or) tags.

Then this action will appear in every branch context menu (either on sidebar on in the commit list).

What do you think?

**Comment by cnovel** [comment]
Not sure about the terminology (maybe Branch custom action can do the trick) but yes what I'm trying to do is pushing a branch, not just a commit. Having the action appearing in the side bar or commit list is good.

**Comment by DanPristupov** [comment]
> Not sure about the terminology (maybe Branch custom action can do the trick)

Yes, I agree. Branch custom action should be clearer.

(In git both branches and tags are references)

OK, I'm adding this to my todo list. I will report ETA later.

**Comment by DanPristupov** [comment]
I just released 1.34 with the support of branch custom actions. I'm looking forward to hear your feedback.

**Comment by cnovel** [comment]
Hi Dan, sorry for the delay. I've just tested it and it's perfectly working. UI is great too. Thanks a lot!
